### PR TITLE
[aws-sdk-cpp] [openssl] Fix iOS build

### PR DESCRIPTION
**Pull request by ahmedyarub**
Don't pass the the compiler to OpenSSL when building for iOS. That would end generating a MacOS library which would cause linking errors when the libraries are used. The OpenSSL configuration scripts know how to detect the target OS and compile using the correct command, example: `xcrun -sdk iphonesimulator cc`. Note that this change should fix building for all iOS targets, including both simulators and actual devices.
Add try_compile() parameters required to test the features of curl when cross-compiling AWS SDK for C++.

- #### What does your PR fix?  
  Fixes building OpenSSL and AWS SDK for C++ for iOS.

- #### Which triplets are supported/not supported? Have you updated the [CI baseline](https://github.com/microsoft/vcpkg/blob/master/scripts/ci.baseline.txt)?  
  All iOS triplets

- #### Does your PR follow the [maintainer guide](https://github.com/microsoft/vcpkg/blob/master/docs/maintainers/maintainer-guide.md)?  
  Yes

- #### If you have added/updated a port: Have you run `./vcpkg x-add-version --all` and committed the result?  
  Yes

**If you are still working on the PR, open it as a Draft: https://github.blog/2019-02-14-introducing-draft-pull-requests/**

### CODE REVIEW

File: versions/a-/aws-sdk-cpp.json [hunk 1]
```diff
@@ -1,5 +1,10 @@
 {
   "versions": [
+    {
+      "git-tree": "2795f006adfb732d610a140ef9eb036f053a5997",
```
Comment (JackBoosY) [inline versions/a-/aws-sdk-cpp.json:4]:
```suggestion
      "git-tree": "9c60baa2cff0bbfc024a8241b96f6685f4e083d3",
```

File: versions/o-/openssl.json [hunk 2]
```diff
@@ -1,5 +1,10 @@
 {
   "versions": [
+    {
+      "git-tree": "2797d49ed1aa47fd6f37965c3db7dc48ca64d1f5",
```
Comment (JackBoosY) [inline versions/o-/openssl.json:4]:
```suggestion
      "git-tree": "3388a88872808e9926c9d94a702bca7e9f2736f9",
```

**Review by JackBoosY — CHANGES_REQUESTED** [review]

**Comment by JackBoosY** [comment]
I think you should wait for my PR https://github.com/microsoft/vcpkg/pull/17341 merge.

**Review by JackBoosY — APPROVED** [review]

File: versions/a-/aws-sdk-cpp.json [hunk 3]
```diff
@@ -1,5 +1,10 @@
 {
   "versions": [
+    {
+      "git-tree": "422c3181effad9596e2a70905d357e39ec33d5cc",
```
Comment (JackBoosY) [inline versions/a-/aws-sdk-cpp.json:4]:
```suggestion
      "git-tree": "c23d66191b235f6f388187395b62f38b8e6e35f5",
```

File: versions/o-/openssl.json [hunk 4]
```diff
@@ -1,5 +1,10 @@
 {
   "versions": [
+    {
+      "git-tree": "f64c492c931c3cb248575a39ae92645392564ad3",
```
Comment (JackBoosY) [inline versions/o-/openssl.json:4]:
```suggestion
      "git-tree": "01fede48f331d05f2feb591061d9e5c1bc076fb0",
```

File: ports/openssl/unix/CMakeLists.txt [hunk 5]
```diff
@@ -143,37 +146,50 @@ else()
...
-        ${PLATFORM}
-        "--prefix=${CMAKE_INSTALL_PREFIX}"
-        "--openssldir=/etc/ssl"
-        ${CFLAGS}
-    COMMAND "${CMAKE_COMMAND}" "-DDIR=${BUILDDIR}" -P "${CMAKE_CURRENT_LIST_DIR}/remove-deps.cmake"
-    VERBATIM
-    WORKING_DIRECTORY "${BUILDDIR}"
-)
+if(IOS)
+    add_custom_command(
```
Comment (ras0219-msft) [inline ports/openssl/unix/CMakeLists.txt:150]: This call should be combined with the main `add_custom_command()` with only the needed parts changed.

For example, `DISABLES` could be set in an if/else chain.

**Review by ras0219-msft — CHANGES_REQUESTED** [review]
Thanks for the PR!

Generally, this smells like we simply aren't correctly propagating flags to the build -- we're missing some flags from the big list on lines 50-77 or so.

This file comes from a time before we had https://github.com/microsoft/vcpkg/blob/master/scripts/cmake/vcpkg_internal_get_cmake_vars.cmake, which is the new way of approaching this problem. See https://github.com/microsoft/vcpkg/blob/master/scripts/get_cmake_vars/CMakeLists.txt for the "modern" version.

I'm not opposed to this stop-gap for now as long as it's better integrated with the existing behavior, but the long term goal will be to move to `vcpkg_internal_get_cmake_vars()`.

**Comment by ahmedyarub** [comment]
> Thanks for the PR!
> 
> Generally, this smells like we simply aren't correctly propagating flags to the build -- we're missing some flags from the big list on lines 50-77 or so.
> 
> This file comes from a time before we had https://github.com/microsoft/vcpkg/blob/master/scripts/cmake/vcpkg_internal_get_cmake_vars.cmake, which is the new way of approaching this problem. See https://github.com/microsoft/vcpkg/blob/master/scripts/get_cmake_vars/CMakeLists.txt for the "modern" version.
> 
> I'm not opposed to this stop-gap for now as long as it's better integrated with the existing behavior, but the long term goal will be to move to `vcpkg_internal_get_cmake_vars()`.

Appreciate the feedback @ras0219-msft . Please give me a couple of days to redo this PR and the Emscripten/WASM32. I'll make the suggested modernization and test on Windows, Ubuntu 20.04 (WSL), MacOS Big Sur, latest Android NDK, iOS and Emscripten SDK on WSL.

**Comment by ahmedyarub** [comment]
> Thanks for the PR!
> 
> Generally, this smells like we simply aren't correctly propagating flags to the build -- we're missing some flags from the big list on lines 50-77 or so.
> 
> This file comes from a time before we had https://github.com/microsoft/vcpkg/blob/master/scripts/cmake/vcpkg_internal_get_cmake_vars.cmake, which is the new way of approaching this problem. See https://github.com/microsoft/vcpkg/blob/master/scripts/get_cmake_vars/CMakeLists.txt for the "modern" version.
> 
> I'm not opposed to this stop-gap for now as long as it's better integrated with the existing behavior, but the long term goal will be to move to `vcpkg_internal_get_cmake_vars()`.

I thought that this method was not supposed to be used by us port devs, right? I tried using vcpkg_configure_make() but it is not compatible at all with OpenSSL and I had to close the PR.

**Comment by JackBoosY** [comment]
Depends on https://github.com/microsoft/vcpkg/pull/17983.

**Review by JackBoosY — APPROVED** [review]

**Comment by strega-nil-ms** [comment]
@ahmedyarub could you rebase onto latest?

**Comment by ahmedalnuaimi** [comment]
> @ahmedyarub could you rebase onto latest?

Done.

**Comment by ahmedalnuaimi** [comment]
Please hold on merging I need to make more tests since the rebase...

**Comment by ahmedyarub** [comment]
> @ahmedyarub could you rebase onto latest?

Please hold on merging I need to make more tests since the rebase...

**Comment by ahmedalnuaimi** [comment]
OK tested both on iOS and Emscripten on Ubuntu. Fixed a couple of bugs and not it should be good to go.

**Comment by ahmedyarub** [comment]
OK tested both on iOS and Emscripten on Ubuntu. Fixed a couple of bugs and not it should be good to go.

**Review by strega-nil-ms — APPROVED** [review]

**Comment by strega-nil-ms** [comment]
Thanks @ahmedyarub :)

**Pull request merged**